crates/autopilot/src/run_loop.rs

```diff
@@ -107,77 +113,115 @@ impl RunLoop {
             trusted_tokens,
             in_flight_orders: Default::default(),
             liveness,
+            readiness,
             maintenance,
             competition_updates_sender,
             winner_selection: winner_selection::Arbitrator { max_winners, weth },
         }
     }
 
-    pub async fn run_forever(self) -> ! {
+    pub async fn run_forever(self, mut control: ShutdownController) {
         Maintenance::spawn_cow_amm_indexing_task(
             self.maintenance.clone(),
             self.eth.current_block().clone(),
         );
+
         let mut last_auction = None;
         let mut last_block = None;
+
         let self_arc = Arc::new(self);
-        loop {
-            let auction = self_arc
-                .next_auction(&mut last_auction, &mut last_block)
-                .await;
-            if let Some(auction) = auction {
+        let mut leader = self_arc
+            .persistence
+            .leader("autopilot_startup".to_string())
+            .await;
+
+        let mut was_leader = false;
+
+        while !control.should_shutdown() {
+            let is_leader = leader.try_acquire().await.unwrap_or_else(|err| {
+                tracing::warn!(error=%err, "failed to become leader");
```

Comment:
I think this pattern is used everywhere.
```suggestion
                tracing::warn!(?err, "failed to become leader");
```
Also, that is a pretty critical error, no? In case autopilot can't become a leader for some unexpected error, this blocks the whole protocol, right? Would the `error` log level be more appropriate for that? Or maybe even creating a Prometheus metric with alerting.

Reply:
Good catch! I will introduce a metric for that.